lib/rules/aria-allowed-role.json

```diff
@@ -0,0 +1,18 @@
+{
+	"id": "aria-allowed-role",
```

Comment:
Thie rule has a lot of overlap with the aria-dpub-role-fallback rule. We may have to exclude the dpub roles from this rule. I'd like to know how the two relate. This rule might actually cover everything the DPUB rule does - if that's the case, we should consider deprecating the DPUB rule.

Reply:
I changed the check in the `aria-dpub-role-fallback` rule. The check for this rule is `implicit-role-fallback` which seems to do certain checks covered in `aria-allowed-role` rule (this one). I swapped out the check on `aria-dpub-role-fallback` to use `aria-allowed-role` and the tests seems to pass. So I think we can deprecate the old one.

Reply:
I'll reach out to @rdeltour to talk about how we want to handle DPUB roles in this rule.

Reply:
I haven't reviewed the PR details, but what would be the rationale for keeping the rule separate? If `aria-allowed-role` covers everything that `aria-dpub-role-fallback` does, then I would be in favor of deprecating the latter.

If you explicitly want the defaults to _not_ include DPUB roles, then yes sharing the underlying checks sounds like a good simplification :-)

Reply:
I did a switcheroo of the checks for these rules. and a few integration tests break and need some re-work. I have created an issue for this, so this deprecation can be handled in as a separate piece of work, than colliding with that this PR is meant to be. 

Issue: https://github.com/dequelabs/axe-core/issues/1033